Pm 28 create a function to save and read images

dotte-osu opened the pull request:
Added image upload functionality and the image to /pet/petID page.
Example: 
![image](https://user-images.githubusercontent.com/71732016/153800566-b3454ac6-dda2-4a27-9fd9-8605bc6dbd34.png)

dotte-osu committed:
added image handling

dotte-osu committed:
finished fileupload

dotte-osu committed:
Merge branch 'main' into PM-28-create-a-function-to-save-images-to-backend

dotte-osu committed:
fixed return content and added picture to pet

xHoudek commented on src/components/PetForm/PetForm.js:
```diff
@@ -269,6 +289,7 @@ export default class PetForm extends Component {
           />
           <label className='form-check-label' htmlFor='houseTrained'>House Trained</label>
         </FormGroup>
+        <FilesUploadComponent id='picture' onChange={this.onFileChange} required />
```
I feel like the image upload gets lost under the checkboxes. What do you think about moving it up to somewhere around the description? There should also be a `<FormGroup>` around this element for consistency, and it should be removed from the `FilesUploadComponent` class

dotte-osu replied:
That's a good idea. I moved it under description.
![image](https://user-images.githubusercontent.com/71732016/153918573-d4833a81-9439-4382-b97e-073eb3953a05.png)

xHoudek commented on src/components/FileUpload/FileUpload.js:
```diff
@@ -0,0 +1,19 @@
+export default class FilesUploadComponent extends Component {
+  render() {
+    return (
+      <div className="container">
```
I don't think we want the container/row divs or the `<FormGroup>` element here. In `Utils.js`, the components basically have just their input:

```
export function Checkbox({ className, ...props }) {
  return (
    <input className={['form-check-input', className].join(' ')} type='checkbox' {...props} />
  );
}
```

I think this could be similar to that, with just the label, form-group div, and file input in there. And then we could put a `<FormGroup>` around `<FilesUploadComponent>` in `PetForm.js`

dotte-osu replied:
Thanks! Fixed it!

xHoudek commented on src/routes/PetPage/PetPage.js:
```diff
@@ -72,7 +72,7 @@ export default function PetPage () {
     // TODO
```
I think this `//TODO` can be removed, unless there is still more work to do

dotte-osu replied:
removed!

xHoudek reviewed (changes requested):
Just some quick code organization on this one. Looks good though! I wasn't able to test because of the bug I created ([PM-28](https://purrfect-match-osu.atlassian.net/jira/software/projects/PM/boards/1?selectedIssue=PM-38)), but will test it out before final approval!

xHoudek commented:
After testing, looks like there are white bars on the side of the picture section on the pet page. Images are also stretched which isn't ideal I think the best behavior would be to fit the window size to match the resolution of the image, and have a bit of the corner cut off so the image fills out the entire window.

Also, pets page gets broken pretty badly when a large-sized image is added(zoomed out a bit for clarity):
![image](https://user-images.githubusercontent.com/46770903/153897493-c5d76727-01f7-4cc1-9768-73a2dc63377d.png)

I think it's out of the scope of this PR to fix these things, but we should add bugfixes to Jira once this story is approved

dotte-osu commented:
Is this the white bar that you are talking about? 
![image](https://user-images.githubusercontent.com/71732016/153920716-abb35b02-83cd-453e-bcb6-af54b7db912f.png)

I completely forgot to check the Pets page. Thanks! And yes, it's horribly broken. I added a class for now but it doesn't make it any better. 
![image](https://user-images.githubusercontent.com/71732016/153920921-47681f85-b18c-481c-8708-1beb5a4860bc.png)

It seems like I need to add image resizer or something. (I have no ide how!!!) I will create a bugfix to Jira and work on it next. 

Other than that, I addressed the code comments.

dotte-osu committed:
addressed code comment

xHoudek commented:
Yeah that is the white bar I was talking about. I'm hoping we can accomplish it by using max-length and max-width CSS properties, but it's probably trickier than that. Code changes look good!

dotte-osu committed:
fixed the image url

dotte-osu commented:
@xHoudek I changed to store only the relative path in the backend and added the `HOSTNAME` in the frontend. It's a tiny change. It would be nice if you can re-approve it. Thanks!
